weather/weather.go

```diff
@@ -24,20 +25,30 @@ type Parser struct {
 	solarRadiation *prometheus.GaugeVec
 	rainIn         *prometheus.GaugeVec
 	ultraviolet    *prometheus.GaugeVec
+	lightning      *prometheus.GaugeVec
+	stationtype    *prometheus.GaugeVec
 }
 
-func NewParser(name string, factory *promauto.Factory) *Parser {
+func NewParser(name string, prefix string, factory *promauto.Factory) *Parser {
+	metric_prefix := ""
+	if prefix != "" {
+		metric_prefix = prefix + "_"
+	}
+
```

Comment:
Looking at `prometheus.GaugeOpts`, it seems you can add a `Namespace` which will get prepended to the metric name with an underscore. Perhaps you should use that instead.

BTW, why do you want a prefix?